### PR TITLE
Migrate the `DeployedRoboJavaModulePlugin` plugin in `buildSrc` to Kotlin

### PR DESCRIPTION
**Pull request by MGaetan89**
This PR converts `DeployedRoboJavaModulePlugin` to Kotlin.

This continues the work of #9344.

### CODE REVIEW

File: buildSrc/src/main/java/org/robolectric/gradle/DeployedRoboJavaModulePlugin.kt [hunk 1]
```diff
@@ -0,0 +1,152 @@
...
+        javadocOptions.noTimestamp(true)
+        javadocOptions.header = extraNavItem
+        javadocOptions.footer = extraNavItem
+        javadocOptions.bottom =
+          """
+          <link rel="stylesheet" href="https://robolectric.org/assets/css/main.css">
+          """
+            .trimIndent()
+        // TODO Restore the following
+        // javadocOptions.version = projectVersion
```
Comment (MGaetan89) [inline buildSrc/src/main/java/org/robolectric/gradle/DeployedRoboJavaModulePlugin.kt:71]: I didn't find any property to set the version, and the generated Javadoc still has the version displayed in the top right corner. So I'm wondering if this was actually doing anything 🤔

Comment (MGaetan89) [reply buildSrc/src/main/java/org/robolectric/gradle/DeployedRoboJavaModulePlugin.kt:71]: If it's unnecessary, I can remove this TODO before this PR gets merged.

Comment (hoisie) [reply buildSrc/src/main/java/org/robolectric/gradle/DeployedRoboJavaModulePlugin.kt:71]: No idea what that's for, I think it's fine to delete it

Comment (MGaetan89) [reply buildSrc/src/main/java/org/robolectric/gradle/DeployedRoboJavaModulePlugin.kt]: Thanks, I've removed the TODO then

**Pull request merged**